packages/Telephony/Common/CommandDialog{T}.cs

```diff
@@ -0,0 +1,146 @@
+
+            // We are expecting a command result with the same name as our current CommandName.
+            if (activity.Type == ActivityTypes.CommandResult
+                && activity.Name == this.CommandName.GetValue(dc.State))
+            {
+                // TODO: correlate command id before handling it.
+
+                var commandResult = TelephonyExtensions.GetCommandResultValue(activity);
+
+                if (commandResult?.Error != null)
+                {
+                    throw new ErrorResponseException($"{commandResult.Error.Code}: {commandResult.Error.Message}");
+                }
+
+                return await dc.EndDialogAsync(cancellationToken: cancellationToken).ConfigureAwait(false);
+            }
+
+            // This activity was not the command result we were expecting. Mark as waiting and end the turn.
+            return new DialogTurnResult(DialogTurnStatus.Waiting);
+        }
+
+        /// <inheritdoc/>
+        protected override async Task<bool> OnPreBubbleEventAsync(DialogContext dc, DialogEvent e, CancellationToken cancellationToken)
+        {
+            if (e.Name == DialogEvents.ActivityReceived && dc.Context.Activity.Type == ActivityTypes.Message)
+            {
+                // Ask parent to perform recognition
+                await dc.Parent.EmitEventAsync(AdaptiveEvents.RecognizeUtterance, value: dc.Context.Activity, bubble: false, cancellationToken: cancellationToken).ConfigureAwait(false);
+
+                // Should we allow interruptions
+                var canInterrupt = true;
+                if (this.AllowInterruptions != null)
+                {
+                    var (allowInterruptions, error) = this.AllowInterruptions.TryGetValue(dc.State);
+                    canInterrupt = error == null && allowInterruptions;
+                }
+
+                // Stop bubbling if interruptions ar NOT allowed
+                return !canInterrupt;
+            }
+
+            return false;
+        }
+
+        private static Activity CreateCommandActivity<TValue>(ITurnContext turnContext, T data, string name)
+        {
+            if (turnContext == null)
+            {
+                throw new ArgumentNullException(nameof(turnContext));
+            }
+
+            if (string.IsNullOrEmpty(name))
+            {
+                throw new ArgumentNullException(nameof(name));
+            }
+
+            var commandActivity = new Activity(type: ActivityTypes.Command);
+
+            commandActivity.Name = name;
+
+            var commandValue = new CommandValue<T>()
+            {
+                CommandId = Guid.NewGuid().ToString(),
+                Data = data,
+            };
+
+            commandActivity.Value = commandValue;
+
+            commandActivity.From = turnContext.Activity.From;
+            
+            // TODO: Check with SDK crew
+            //commandActivity.RelatesTo = turnContext.Activity.GetConversationReference();
```

Comment:
Remove this commented line + comment pls, we don't need this right now.

Reply:
Done